### DISCUSSION TITLE
[4.0] Permissions for Individual Template Styles (for Group Assignment)

### DISCUSSION THREAD

**Discussion by orware** [joomla/joomla-cms#39244] [2. Cold Features]
### Steps to reproduce the issue

I've primarily managed our campus website on my own for the past 10 years and the last 8 or so we've been using Joomla 1.6+ (currently on the latest version of Joomla 3.9.x) and there's been a request to maybe use a separate template for a part of the site.

The only way I've managed to keep things maintainable over the long run with our various users has been to keep the interface simplified, so I currently am using the Joomla ACL so that our staff by default have the ability to login to the backend but wouldn't be able to see any thing at all (this is provided by a Basic Backend Access Group I've setup). Most users would never do that on their own however, so usually the first time they login they've already been assigned a User Group I've created that grants them access to the Categories on the website that they need access to.

So I don't normally try and grant anything else (no Menu/Module/Template related permissions).

With the request that's come up recently, I'm taking a look at the available permissions for the Templates area, and it only appears to have a component-level permission, but does not appear to have a Permissions tab available for any of the Template Styles, which would allow me to create a special template style just for this area and then grant the users access to it.

I believe not too long ago permissions were extended on the Menus side of things (I can't remember if the Modules area was included in that set of updates or if the Permissions have been available there for a longer period of time), so I think what would need to be done is similar, just applied to the Templates component so Permissions can be assigned for each Template Style too.

### Expected result

Have an ability to assign Permissions at the individual Template Style level.

### Actual result

No Permissions tab is available at the Template Style Level.

### Additional comments

This could potentially be related to my now somewhat old Pull Request located here:
https://github.com/joomla/joomla-cms/pull/20508

And Issue here (which was closed after I submitted the Pull Request above last May):
https://github.com/joomla/joomla-cms/issues/20506

I've been maintaining the patches there offline for our own install of Joomla so I have patches that are current for Joomla 3.9.5, but one thing I noticed when I was running my tests above is that the Menus and Modules components (and likely Templates too) exhibit the same behavior I fixed in the patches above for com_content / com_categories so that when users have been assigned a limited access group, the other rows are automatically filtered out and not even displayed on the page (the default behavior in Joomla is to continue displaying the rows a user doesn't have access to and just simply grays them out...this I think is due to the way things used to work prior to the introduction of the Joomla ACL, since back then, and now with the default user groups built-in to Joomla...if you have backend access you basically can see the entirety of these areas...with the Joomla ACL now available though, I think it makes a lot of sense to improve the UI by not displaying those unneeded rows to users that don't have access to the records anyway).

**Comment by ghost** [comment]
@HLeithner can you please comment?

**Comment by HLeithner** [comment]
It's a feature request and would be j4 related, maybe @orware could help building a PR for J4.

@wilsonge would you have a look?

**Comment by wilsonge** [comment]
Seems like an interesting concept to add asset table entries for template styles. Would also make sense in multilang sites so that only the language admins could edit their template styles. I'd be interested in seeing a PR.

**Comment by Gnevyshev** [comment]
I join this question! I want to be able to set the access level for the templates!

**Comment by alikon** [comment]
Closing this due to not receiving required information to determine if this is a bug or not. If you feel this still needs review, please open a new tracker entry with as much information as possible to ensure it can be reviewed properly

**Comment by brianteeman** [comment]
@alikon This was not a bug report it was a valid feature request

**Comment by ceford** [comment]
It is not clear what the users will be allowed to do with their permissions. At the moment, users can choose a style from those available. If you are creating the style they should be able to use it. I should imagine that allowing multiple users to change styles could lead to bun fights. <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/24560">issues.joomla.org/tracker/joomla-cms/24560</a>.</sub>

**Comment by alikon** [comment]
as it is not a bug but a 

> a valid feature request

let's move it to discussion

**Reply by crystalenka** [reply]
Moved back to issues.

**Reply by crystalenka** [reply]
@wojtekxtx Unfortunately once an issue has been converted to a discussion, it cannot be directly converted back. There is only "create issue from discussion", which is what I did.
